dispatchers_test.go

```diff
@@ -90,6 +91,129 @@ func TestDispatchCopy(t *testing.T) {
 	}
 }
 
+func TestDispatchCopyHeredoc(t *testing.T) {
+	mybuilder := Builder{
+		RunConfig: docker.Config{
+			WorkingDir: "/root",
+			Cmd:        []string{"/bin/sh"},
+			Image:      "alpine",
+		},
+	}
+	args := []string{"<<robots.txt <<humans.txt", "/test/"}
+	flagArgs := []string{}
+	original := "COPY <<robots.txt <<humans.txt /test/"
+	heredocs := []buildkitparser.Heredoc{{Name: "robots.txt", FileDescriptor: 0, Chomp: false, Content: "(robots content)"}, {Name: "humans.txt", FileDescriptor: 0, Chomp: false, Content: "(humans content)"}}
+	if err := dispatchCopy(&mybuilder, args, nil, flagArgs, original, heredocs); err != nil {
+		t.Errorf("dispatchCopy error: %v", err)
+	}
+	expectedPendingCopies := []Copy{
+		{
+			From:     "",
+			Src:      []string{"<<robots.txt <<humans.txt"},
+			Dest:     "/test/", // destination must contain a trailing slash
+			Download: false,
+			Chown:    "",
+			Chmod:    "",
+			Files:    []File{{Name: "robots.txt", Data: "(robots content)"}, {Name: "humans.txt", Data: "(humans content)"}},
+		},
+	}
+	if !reflect.DeepEqual(mybuilder.PendingCopies, expectedPendingCopies) {
+		t.Errorf("Expected %v, got %v\n", expectedPendingCopies, mybuilder.PendingCopies)
+	}
+}
+
+func TestDispatchAddHeredoc(t *testing.T) {
+	mybuilder := Builder{
+		RunConfig: docker.Config{
+			WorkingDir: "/root",
+			Cmd:        []string{"/bin/sh"},
+			Image:      "alpine",
+		},
+	}
+
+	args := []string{"<<EOF", "/index.html"}
+	flagArgs := []string{}
+	original := "ADD <<EOF /index.html"
+	heredocs := []buildkitparser.Heredoc{{Name: "EOF", Expand: true, Content: "(your index page goes here)"}}
+	if err := add(&mybuilder, args, nil, flagArgs, original, heredocs); err != nil {
+		t.Errorf("add error: %v", err)
+	}
+
+	expectedPendingCopies := []Copy{
+		{
+			From:     "",
+			Src:      []string{"<<EOF"},
+			Dest:     "/index.html", // destination must contain a trailing slash
```

Comment:
This comment is a bit confusing, since the value for this field does not contain a trailing slash.

Reply:
removed comment it was copied from another block